react/src/components/FolderInvitationResponseModalOpener.tsx

```diff
@@ -0,0 +1,42 @@
+import { useVFolderInvitationsValue } from '../hooks/useVFolderInvitations';
+import React from 'react';
+import { useQueryParam, BooleanParam } from 'use-query-params';
+
+const FolderInvitationResponseModal = React.lazy(
+  () => import('./FolderInvitationResponseModal'),
+);
+
+const FolderInvitationResponseModalOpener = () => {
+  const [isInvitationOpen, setIsInvitationOpen] = useQueryParam(
+    'invitation',
+    BooleanParam,
+  );
+  const { count } = useVFolderInvitationsValue();
+
+  return (
+    <FolderInvitationResponseModal
+      open={!!isInvitationOpen}
+      onRequestClose={(success) => {
+        if (success) {
+          if (count === 1) {
+            setIsInvitationOpen(null, 'replaceIn');
```

Comment:
We can reduce if-statement. 

```suggestion
            if (!success || count === 1) {
              setIsInvitationOpen(null, 'replaceIn');
            }
```